### PR TITLE
wireless/gs2200m: replace the preamble scratch to union 

### PR DESCRIPTION
**Pull request by anchao**
## Summary

wireless/gs2200m: replace the preamble scratch to union 

## Impact

fix the preamble scratch buffer issue

## Testing

N/A, I do not have the wifi module....

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@masayuki2009 could you try the change?

File: wireless/gs2200m/gs2200m_main.c [hunk 1]
```diff
@@ -389,17 +406,17 @@ static int usrsock_request(int fd, FAR struct gs2200m_s *priv)
       return -EIO;
     }
 
-  assert(handlers[com_hdr->reqid].hdrlen < sizeof(hdrbuf));
+  assert(handlers[com_hdr->reqid].hdrlen < sizeof(req));
```
Comment (masayuki2009) [inline wireless/gs2200m/gs2200m_main.c:409]: @anchao 

I found that the assertion happend here.
I confirmed the following condition works.

```
  assert(handlers[com_hdr->reqid].hdrlen <= sizeof(req));
```

Comment (anchao) [reply wireless/gs2200m/gs2200m_main.c:409]: Thanks for your quick reply, I will update here, will the spresense works properly if this change included?

Comment (masayuki2009) [reply wireless/gs2200m/gs2200m_main.c]: > Thanks for your quick reply, I will update here, will the spresense works properly if this change included?

Yes. (I'm still checking the ntp client stack corruption https://github.com/apache/incubator-nuttx-apps/pull/957 but the issue does not relate to this PR)

**Review by masayuki2009 — APPROVED** [review]

**Pull request merged**